storage/badger/operation/heights.go

```diff
@@ -38,6 +38,12 @@ func RetrieveSealedHeight(height *uint64) func(*badger.Txn) error {
 	return retrieve(makePrefix(codeSealedHeight), height)
 }
 
+// SetLastCompleteBlockHeight inserts the last full block height if it is not already set.
+// Calling this function multiple times is a no-op and returns no expected errors.
+func SetLastCompleteBlockHeight(height uint64) func(*badger.Txn) error {
+	return SkipDuplicates(update(makePrefix(codeLastCompleteBlockHeight), height))
```

Comment:
I think update will fail for the first operation, [because the key does exist yet.](https://github.com/onflow/flow-go/blob/master/storage/badger/operation/common.go#L125)

Reply:
good catch I updated it to insert instead https://github.com/onflow/flow-go/pull/2974/commits/9e396d8e6097b0f8ecf7cec97181aac118f6a530